src/Features/CSharp/Portable/ImplementInterface/AbstractChangeImplementionCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,267 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ImplementInterface
+{
+    // A mapping from a concrete member to the interface members it implements.
+    using MemberImplementationMap = OrderedMultiDictionary<ISymbol, ISymbol>;
```

Comment:
Why the orderedness of it?

Reply:
(if you don't, then I think OrderedMultiDictionary can be deleted, as I don't see any uses in the compiler layer. I thought I had already deleted this for good....)

Reply:
so it's def necessary (and i'll document).  We need things to be ordered so we process things in a consistent order (i.e. when a member implements multiple interface methods). When i wrote this oriinally, it wasn't ordered, and you could get inconsistent test results.

Reply:
doc'ed.

Reply:
Thanks! I was thinking that OrderedMultiDictionary only managed ordering of the keys, but I guess the regular one just uses a HashSet for the values too.